From 9538e1e17c0bd6c5c5e7f4b4569095a4d67a035b Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Thu, 11 Aug 2016 13:36:09 -0700
Subject: [PATCH 1718/2748] HDFS-8600.
 TestWebHdfsFileSystemContract.testGetFileBlockLocations
 fails in branch-2.7. (Contributed by Arpit
 Agarwal)

(cherry picked from commit a1810d75f05418115c660f2dc02335504fff4935)

Change-Id: Ie3bc0191128bbc5376309a7b55323746da5383ae
---
 .../java/org/apache/hadoop/hdfs/web/JsonUtil.java  |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/web/JsonUtil.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/web/JsonUtil.java
index b9186fa..f933339 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/web/JsonUtil.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/web/JsonUtil.java
@@ -367,10 +367,9 @@ static DatanodeInfo toDatanodeInfo(final Map<?, ?> m)
 
     // Handle the case of old servers (1.x, 0.23.x) sending 'name' instead
     // of ipAddr and xferPort.
+    int xferPort = getInt(m, "xferPort", -1);
     Object tmpValue = m.get("ipAddr");
     String ipAddr = (tmpValue == null) ? null : (String)tmpValue;
-    tmpValue = m.get("xferPort");
-    int xferPort = (tmpValue == null) ? -1 : (int)(long)(Long)tmpValue;
     if (ipAddr == null) {
       tmpValue = m.get("name");
       if (tmpValue != null) {
-- 
1.7.9.5

